client/lib/domains/index.js

```diff
@@ -58,9 +58,11 @@ function checkDomainAvailability( domainName, onComplete ) {
 			if ( unmappabilityReason ) {
 				errorCode += `_${ unmappabilityReason }`;
 			}
-		} else if ( ! isAvailable && isMappable ) {
+		} else if ( isAvailable === null && isMappable ) {
+			errorCode = 'no_availability_but_mappable';
+		} else if ( isAvailable === false && isMappable ) {
```

Comment:
Sorry, I don't think there will ever be a valid use case for a nullable boolean in JavaScript. The implementation is way too implicit and I can guarantee that someone will rightfully assume a boolean value can only be true or false and hit a bug. There are many solutions to this, we can create another property from the API and use that, we can create a v2 API send fix all the mess with x_but_y cases, we can change the copy to convey this domain is available to map if they own it.